web/src/search/backend.tsx

```diff
@@ -452,3 +452,31 @@ export const highlightCode = memoizeObservable(
         ),
     ctx => `${ctx.code}:${ctx.fuzzyLanguage}:${ctx.disableTimeout}:${ctx.isLightTheme}`
 )
+
+/**
+ * Returns true if search performance and accuracy are limited because this is a
+ * single-node Docker deployment that is configured with more than 100 repositories.
+ */
+export function displayPerformanceWarning(deployType: DeployType): Observable<boolean> {
```

Comment:
```suggestion
export function shouldDisplayPerformanceWarning(deployType: DeployType): Observable<boolean> {
```

Because this doesn't actually _do_ anything to display the warning, it merely queries the backend to find out whether you should display it.